backend/danswer/connectors/freshdesk/connector.py

```diff
@@ -0,0 +1,115 @@
+        self.domain = credentials.get("freshdesk_domain")
+        self.password = credentials.get("freshdesk_password")
+        return None
+
+    def _process_tickets(self, start: datetime, end: datetime) -> GenerateDocumentsOutput:
+        logger.info("Processing tickets")
+        if any([self.api_key, self.domain, self.password]) is None:
+            raise ConnectorMissingCredentialError("freshdesk")
+
+        freshdesk_url = f"https://{self.domain}.freshdesk.com/api/v2/tickets?include=description"
+        response = requests.get(freshdesk_url, auth=(self.api_key, self.password))
+        response.raise_for_status()  # raises exception when not a 2xx response
+
+        if response.status_code!= 204:
+            tickets = json.loads(response.content)
+            logger.info(f"Fetched {len(tickets)} tickets from Freshdesk API")
+            doc_batch: List[Document] = []
+
+            for ticket in tickets:
+                # Convert the "created_at", "updated_at", and "due_by" values to ISO 8601 strings
+                for date_field in ["created_at", "updated_at", "due_by"]:
+                    if ticket[date_field].endswith('Z'):
+                        ticket[date_field] = ticket[date_field][:-1] + '+00:00'
+                    ticket[date_field] = datetime.fromisoformat(ticket[date_field]).strftime("%Y-%m-%d %H:%M:%S")
+
+                # Convert all other values to strings
+                ticket = {
+                    key: str(value) if not isinstance(value, str) else value
+                    for key, value in ticket.items()
+                }
+
+                # Checking for overdue tickets
+                today = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+                ticket["overdue"] = "true" if today > ticket["due_by"] else "false"
+
+                # Mapping the status field values
+                status_mapping = {2: "open", 3: "pending", 4: "resolved", 5: "closed"}
+                ticket["status"] = status_mapping.get(ticket["status"], str(ticket["status"]))
+
+                # Stripping HTML tags from the description field
+                ticket["description"] = self.strip_html_tags(ticket["description"])
+
+                # Remove extra white spaces from the description field
+                ticket["description"] = " ".join(ticket["description"].split())
+
+                # Use list comprehension for building sections
+                sections = self.build_doc_sections_from_ticket(ticket)
+
+                created_at = datetime.fromisoformat(ticket["created_at"])
+                today = datetime.now()
+                if (today - created_at).days / 30.4375 <= 2:
+                    doc = Document(
+                        id=ticket["id"],
+                        sections=sections,
+                        source=DocumentSource.FRESHDESK,
+                        semantic_identifier=ticket["subject"],
+                        metadata={
```

Comment:
im concerned about this metadata. may be worthwhile to double check it to make sure its reasonable